File: vim/custom_script/copyandpay/copyandpay_checkout.py

# Copyright (c) 2015, Frappe Technologies Pvt. Ltd. and Contributors
# License: GNU General Public License v3. See license.txt
from __future__ import unicode_literals
import frappe
from frappe import _
from frappe.utils import flt, cint
import json
from six import string_types


@frappe.whitelist(allow_guest=True)
def make_payment(copyandpay_payment_id, options, reference_doctype, reference_docname, token,checkoutid):
	data = {}

	if isinstance(options, string_types):
		data = json.loads(options)

	data.update({
		"copyandpay_payment_id": copyandpay_payment_id,
		"reference_docname": reference_docname,
		"reference_doctype": reference_doctype,
		"token": token,
		"checkoutid":checkoutid
	})
	
	data =  frappe.get_doc("CopyandPay Settings").create_request(data)
	frappe.db.commit()
	return data

File: vim/custom_script/copyandpay/copyandpay_checkout.js
    
$(document).ready(function(){
    console.log("fhdhdfh")
	(function(e){
		var options = {
			"key": "{{ api_key }}",
			"amount": cint({{ amount }} * 100), // 2000 paise = INR 20
			"name": "{{ title }}",
			"description": "{{ description }}",
			"subscription_id": "{{ subscription_id }}",
			"handler": function (response){
				copyandpay.make_payment_log(response, options, "{{ reference_doctype }}", "{{ reference_docname }}", "{{ token }}","{{ checkout }}");
			},
			"prefill": {
				"name": "{{ payer_name }}",
				"email": "{{ payer_email }}",
				"order_id": "{{ order_id }}"
			},
			"notes": {{ frappe.form_dict|json }}
		};

		var rzp = new CopyandPay(options);
		rzp.open();
		//	e.preventDefault();
	})();
})


frappe.provide('copyandpay');

copyandpay.make_payment_log = function(response, options, doctype, docname, token,checkout){
	$('.copyandpay-loading').addClass('hidden');
	$('.copyandpay-confirming').removeClass('hidden');
    
    
	frappe.call({
		method:"vim/custom_script/copyandpay/copyandpay_checkout.make_payment",
		freeze:true,
		headers: {"X-Requested-With": "XMLHttpRequest"},
		args: {
			"copyandpay_payment_id": response.copyandpay_payment_id,
			"options": options,
			"reference_doctype": doctype,
			"reference_docname": docname,
			"token": token,
            "checkoutid":checkout
		},
		callback: function(r){
			if (r.message && r.message.status == 200) {
				window.location.href = r.message.redirect_to
			}
			else if (r.message && ([401,400,500].indexOf(r.message.status) > -1)) {
				window.location.href = r.message.redirect_to
			}
		}
	})
}
